controllers/pattern_controller.go

```diff
@@ -180,35 +183,32 @@ func (r *PatternReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		}
 	}
 
-	// -- GitOps Subscription
-	targetSub := newSubscription(*qualifiedInstance)
-	_ = controllerutil.SetOwnerReference(qualifiedInstance, targetSub, r.Scheme)
-
-	_, sub := getSubscription(r.olmClient, targetSub.Name, targetSub.Namespace)
-	if sub == nil {
-		err := createSubscription(r.olmClient, targetSub)
-		return r.actionPerformed(qualifiedInstance, "create gitops subscription", err)
-	} else if ownedBySame(targetSub, sub) {
-		// Check version/channel etc
-		// Dangerous if multiple patterns do not agree, or automatic upgrades are in place...
-		err, changed := updateSubscription(r.olmClient, targetSub, sub)
-		if changed {
-			return r.actionPerformed(qualifiedInstance, "update gitops subscription", err)
-		}
-	} else {
-		logOnce("The gitops subscription is not owned by us, leaving untouched")
+	// Deploy openshift-gitops subscription from config map
+	err = r.deployGitopsSubscription(customConfigFile)
+	if err != nil {
+		fmt.Print("Error: ", err)
+		r.logger.Info("Issue deploying openshift-gitops from configmap")
 	}
 
-	logOnce("subscription found")
-
-	// -- GitOps Namespace (created by the gitops operator)
-	if !haveNamespace(r.Client, applicationNamespace) {
-		return r.actionPerformed(qualifiedInstance, "check application namespace", fmt.Errorf("waiting for creation"))
+	// -- GitOps Subscription
+	s, err := r.olmClient.getSubscription(gitopsSubscriptionName, subscriptionNamespace)
+	if err != nil {
+		return r.actionPerformed(qualifiedInstance, "get gitops subscription", err)
+	}
+	if s.Status.InstallPlanRef == nil {
+		return r.actionPerformed(qualifiedInstance, "get gitops installplan", fmt.Errorf("waiting for installplan to be created"))
+	}
+	p, err := r.olmClient.getInstallPlan(s.Status.InstallPlanRef.Name)
+	if err != nil {
+		return r.actionPerformed(qualifiedInstance, "get gitops subscription phase", err)
+	}
+	if p.Status.Phase != operatorv1alpha1.InstallPlanPhaseComplete {
+		return r.actionPerformed(qualifiedInstance, "gitops subscription phase", fmt.Errorf("gitops subscription deployment is not yet complete"))
```

Comment:
Do we get much value out of this?

Reply:
I have restored what we had in our upstream.